### DISCUSSION TITLE
uac2_headset Problems

### DISCUSSION THREAD

**Discussion by FearLabsAudio** [hathach/tinyusb#2905] [Q&A]
I have been attempting to get the uac2_headset example running with mixed results.    

Running windows 11 host,  pico 2 device

I was randomly hitting the previously mentioned hard_assert(hw_data_offset(next_buffer_ptr) <= USB_DPRAM_MAX); assert.  For the most part it would actually work, but the following would cause it:
 - Clearing the windows USB cache and plugging, or running on a new computer
 - opening/closing the sound settings panel or changing the bit depth repeatedly

After updating to the latest commit today, I am no longer hitting that assert.  Volume/Mute changes register, however any time audio playback is started the debug output shows the following:

```
Clock set current freq: 48000
WARN: starting new transfer on already active ep 01

*** PANIC ***

ep 01 was already available
```

Sometimes it will show other ep#'s ( ep 81 is common )

Does anyone have any thoughts?  Thanks much.

**Comment by FearLabsAudio** [comment]
Just to rule out anything, I set up a project from scratch.  

HOST: windows 11 (also same result under Linux Mint)
Device: RP2040/Pico 1 (in case there was any issue with Pico 2 being too new)

Tinyusb library was updated to the latest commit.
PicoSDK 2.1

Here is the output with CFG_TUSB_DEBUG=1

```
Headset running
Get channel 0 volume range (-50, 0, 1) dB
Get channel 1 volume range (-50, 0, 1) dB
Get channel 2 volume range (-50, 0, 1) dB
Clock get 2 freq ranges
Range 0 (44100, 44100, 0)
Range 1 (48000, 48000, 0)
Get channel 1 volume 0 dB
Get channel 1 mute 0
Get channel 1 volume 0 dB
Get channel 1 mute 0
Set channel 1 volume: 0 dB
Set channel 2 volume: 0 dB
Get channel 1 volume 0 dB
Set channel 1 Mute: 0
Get channel 2 volume 0 dB
Set channel 2 Mute: 0
Get channel 1 mute 0
Get channel 1 volume 0 dB
Get channel 1 mute 0
Get channel 1 volume 0 dB
Get channel 1 mute 0
Get channel 1 volume 0 dB
Get channel 1 mute 0
Get channel 1 volume 0 dB
Get channel 1 mute 0
Clock set current freq: 48000
WARN: starting new transfer on already active ep 01

*** PANIC ***

ep 01 was already available
```

For anyone who said they had this example working:  Were you running linux/windows/macos?  Were you running an RP2040 or what device? 

Thanks all.

**Reply by hathach** [reply]
please try latest master, this should be fixed
